nautobot/extras/models/models.py

```diff
@@ -769,28 +771,38 @@ def check_for_conflicts(cls, instance, content_types, payload_url, type_create,
         conflicts = {}
         webhook_error_msg = "A webhook already exists for {action} on {content_type} to URL {url}"
 
-        for content_type in content_types:
-            webhooks = cls.objects.filter(content_types__in=[content_type], payload_url=payload_url)
-            if instance and instance.present_in_database:
-                webhooks = webhooks.exclude(pk=instance.pk)
-
-            existing_type_create = webhooks.filter(type_create=type_create).exists() if type_create else False
-            existing_type_update = webhooks.filter(type_update=type_update).exists() if type_update else False
-            existing_type_delete = webhooks.filter(type_delete=type_delete).exists() if type_delete else False
-
-            if existing_type_create:
-                conflicts.setdefault("type_create", []).append(
-                    webhook_error_msg.format(content_type=content_type, action="create", url=payload_url),
-                )
-
-            if existing_type_update:
-                conflicts.setdefault("type_update", []).append(
-                    webhook_error_msg.format(content_type=content_type, action="update", url=payload_url),
-                )
-
-            if existing_type_delete:
-                conflicts.setdefault("type_delete", []).append(
-                    webhook_error_msg.format(content_type=content_type, action="delete", url=payload_url),
-                )
+        if content_types is None and instance is not None:
+            # This is a PATCH and might not include all relevant data e.g content_types, payload_url or actions
+            # Therefore we get data not available from instance
+            content_types = instance.content_types.all()
+            payload_url = instance.payload_url if payload_url is None else payload_url
+            type_create = instance.type_create if type_create is None else type_create
+            type_update = instance.type_update if type_update is None else type_update
+            type_delete = instance.type_delete if type_delete is None else type_delete
+
+        if content_types is not None:
```

Comment:
When would `content_types`still  be None, or in other words why was it necessary to add this check?

Reply:
`content_types` is not a required field, which means a webhook can be created without `content_types` and a patch on that webhook which do not include a content_type would break.
Its not logical for this to happen but its possible